src/platform_impl/macos/view.rs

```diff
@@ -414,16 +437,44 @@ extern "C" fn selected_range(_this: &Object, _sel: Sel) -> NSRange {
     util::EMPTY_RANGE
 }
 
+fn id_to_string(string: id) -> String {
+    unsafe {
+        let has_attr = msg_send![string, isKindOfClass: class!(NSAttributedString)];
+        let characters = if has_attr {
+            // This is a *mut NSAttributedString
+            msg_send![string, string]
+        } else {
+            // This is already a *mut NSString
+            string
+        };
+        let utf8_sequence =
+            slice::from_raw_parts(characters.UTF8String() as *const c_uchar, characters.len());
+        str::from_utf8_unchecked(utf8_sequence).to_string()
+    }
+}
```

Comment:
- I think this function should be moved to the `util` module, so that it can be re-used from other places
- The function itself should be marked unsafe, because it assumes that the `string` is either an `NSAttributedString` or an `NSString`. (And the function doesn't check if this assumption is true.)

Reply:
You're right. I move this function to `util/` and mark this function unsafe.